controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +201,121 @@ func (r *PolicyAutomationReconciler) Reconcile(
 			} else {
 				log.Info("All clusters are compliant. Doing nothing.")
 			}
+		} else if policyAutomation.Spec.Mode == "everyEvent" {
+			log := log.WithValues("mode", "everyEvent")
+			targetList := common.FindNonCompliantClustersForPolicy(policy)
+			eventTime := time.Now().UTC().Format(time.RFC3339)
+			if len(targetList) > 0 {
+				delayAfterRunSecondsStr := policyAutomation.Spec.DelayAfterRunSeconds
+				delayAfterRunSeconds, err := strconv.Atoi(delayAfterRunSecondsStr)
+				if err == nil {
+					log.Info("Can't read delayAfterRunSeconds, set to default value zero")
+					// if delayAfterRunSeconds is unset or isn't int, set to default value 0
+					delayAfterRunSeconds = 0
+				}
+				statusTemp := policyAutomation.Status
+				if len(statusTemp.ClustersWithEvent) == 0 {
+					log.Info("Creating an Ansible job", "targetList", targetList)
+
+					err = common.CreateAnsibleJob(policyAutomation, r.DynamicClient, "everyEvent", targetList)
+					if err != nil {
+						log.Error(err, "Failed to create the Ansible job")
+
+						return reconcile.Result{}, err
+					}
+					automationStartTime := time.Now().UTC().Format(time.RFC3339)
+
+					eventMap := make(map[string]policyv1beta1.ClusterEventTime)
+					for _, clusterName := range targetList {
+						eventMap[clusterName] = policyv1beta1.ClusterEventTime{
+							AutomationStartTime: automationStartTime,
+							EventTime:           eventTime, // become non compliant timestamp
+						}
+					}
+
+					statusTemp.ClustersWithEvent = eventMap
+					policyAutomation.Status = statusTemp
+
+					err = r.Update(ctx, policyAutomation, &client.UpdateOptions{})
+					if err != nil {
+						log.Error(err, "Failed to update clusters event time in policyAutomation status")
+
+						return reconcile.Result{}, err
+					}
+				} else {
+					eventMap := statusTemp.ClustersWithEvent
+
+					targetListStr := strings.Join(targetList, ",")
```

Comment:
Could you make this a set instead?

For example:
```suggestion
					targets = map[string]bool{}
					for _, target := range targetList {
						targets[target] = true
					}
```


Then you can just do the following to check if it's present, which is a very fast operation:
```go
if targets[clusterName] {
```